From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Javier Martinez Canillas <javierm@redhat.com>
Date: Fri, 11 Jun 2021 13:13:27 +0200
Subject: [PATCH] Don't check for rpmvercmp in librpm

The rpmvercmp() function was moved from librpm to librpmio. The configure
option had some logic to first check if the symbol is in librpm and then
librpmio if this check didn't succeed.

But the logic wasn't working and rpm sorting was always disabled. Instead
of trying to fix this logic, let's just remove since the function already
moved and there's no need to check librpm anymore. Now it's enabled again:

  GRUB2 will be compiled with following components:
  ...
  grub-rpm-sort: Yes
  ...

Signed-off-by: Javier Martinez Canillas <javierm@redhat.com>
---
 configure.ac | 19 +++++--------------
 1 file changed, 5 insertions(+), 14 deletions(-)

diff --git a/configure.ac b/configure.ac
index c7842ec29d8..3c808a72230 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1947,24 +1947,15 @@ if test x"$rpm_sort_excuse" = x ; then
 fi
 
 if test x"$rpm_sort_excuse" = x ; then
-  # Check for rpm library.
-  AC_CHECK_LIB([rpm], [rpmvercmp], [],
-               [rpm_sort_excuse="rpmlib missing rpmvercmp"])
-fi
-
-if test x"$rpm_sort_excuse" = x ; then
-   LIBRPM="-lrpm";
-   AC_DEFINE([HAVE_RPM], [1],
-             [Define to 1 if you have the rpm library.])
-fi
-
-if test x"$LIBRPM" = x ; then
-  # Check for rpm library.
+  # Check for rpmio library.
   AC_CHECK_LIB([rpmio], [rpmvercmp], [],
                [rpm_sort_excuse="rpmio missing rpmvercmp"])
+fi
+
+if test x"$rpm_sort_excuse" = x ; then
    LIBRPM="-lrpmio";
    AC_DEFINE([HAVE_RPMIO], [1],
-             [Define to 1 if you have the rpm library.])
+             [Define to 1 if you have the rpmio library.])
 fi
 
 AC_SUBST([LIBRPM])
